jerry-core/jerry.c

```diff
@@ -1848,7 +1848,8 @@ snapshot_add_compiled_code (ecma_compiled_code_t *compiled_code_p, /**< compiled
 
   snapshot_last_compiled_code_offset = snapshot_buffer_write_offset;
 
-  compiled_code_map_entry_t *new_entry = (compiled_code_map_entry_t *) mem_pools_alloc ();
+  compiled_code_map_entry_t *new_entry =
+  (compiled_code_map_entry_t *) mem_heap_alloc_block (sizeof (compiled_code_map_entry_t));
```

Comment:
But the proposed solution looks ugly. Introduce a two parameter macro for allocating, or improve the style checker.